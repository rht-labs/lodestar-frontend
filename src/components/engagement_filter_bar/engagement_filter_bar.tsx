import React from 'react';
import {
  TextInput,
  Button,
  ButtonVariant,
  InputGroup,
  Flex,
  FlexItem,
} from '@patternfly/react-core';
import { EngagementFilter } from '../../schemas/engagement_filter';
import { SortSelect } from './components/sort_select';
import { EngagementStatusSelect } from './components/engagement_status_select';
import { SearchIcon } from '@patternfly/react-icons';

export interface EngagementFilterProps {
  onChange: (filter: EngagementFilter) => void;
  filter: EngagementFilter;
}

export function EngagementFilterBar({
  onChange,
  filter,
}: EngagementFilterProps) {
  const { searchTerm = '' } = filter ?? {};
  return (
    <Flex justifyContent={{ default: 'justifyContentFlexStart' }}>
      <Flex
        style={{ marginTop: '2rem' }}
        direction={{ default: 'column', md: 'row' }}
      >
        <FlexItem grow={{ default: 'grow' }}>
          <InputGroup>
            <Button
              variant="control"
              aria-label="search button for search input"
            >
              <SearchIcon />
            </Button>
            <TextInput
              style={{ minWidth: '15rem' }}
              aria-label="Search for an engagement"
              value={searchTerm}
              onChange={searchTerm => onChange({ ...filter, searchTerm })}
              placeholder="Search for an engagement"
              data-cy={'search_input'}
            />
          </InputGroup>
        </FlexItem>
        <FlexItem>
          <EngagementStatusSelect onChange={onChange} filter={filter} />
        </FlexItem>
        <FlexItem>
          <SortSelect onChange={onChange} filter={filter} />
        </FlexItem>
        <FlexItem>
          <Button
            variant={ButtonVariant.secondary}
            onClick={() => onChange({})}
            data-cy={'reset_button'}
          >
            Reset
          </Button>
        </FlexItem>
      </Flex>
    </Flex>
  );
}
