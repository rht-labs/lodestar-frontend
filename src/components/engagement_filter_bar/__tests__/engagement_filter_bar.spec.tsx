import React from 'react';
import { render } from '@testing-library/react';
import { EngagementFilterBar } from '../engagement_filter_bar';
describe('Engagement Navigation ', () => {
  test('matches the snapshot', () => {
    expect(
      render(<EngagementFilterBar filter={{}} onChange={() => {}} />)
    ).toMatchSnapshot();
  });
});
