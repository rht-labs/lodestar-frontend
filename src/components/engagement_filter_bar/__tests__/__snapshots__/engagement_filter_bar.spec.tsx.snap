// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement Navigation  matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-l-flex pf-m-justify-content-flex-start"
      >
        <div
          class="pf-l-flex pf-m-column pf-m-row-on-md"
          style="margin-top: 2rem;"
        >
          <div
            class="pf-m-grow"
          >
            <div
              class="pf-c-input-group"
            >
              <button
                aria-disabled="false"
                aria-label="search button for search input"
                class="pf-c-button pf-m-control"
                data-ouia-component-id="0"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    transform=""
                  />
                </svg>
              </button>
              <input
                aria-invalid="false"
                aria-label="Search for an engagement"
                class="pf-c-form-control"
                data-cy="search_input"
                placeholder="Search for an engagement"
                style="min-width: 15rem;"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=""
          >
            <div
              class="pf-c-input-group"
            >
              <button
                aria-disabled="false"
                aria-label="search button for search input"
                class="pf-c-button pf-m-control"
                data-ouia-component-id="1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
                    transform=""
                  />
                </svg>
              </button>
              <div
                class="pf-c-select"
                data-ouia-component-id="2"
                data-ouia-component-type="PF4/Select"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby=" filter_dropdown"
                  class="pf-c-select__toggle"
                  id="filter_dropdown"
                  type="button"
                >
                  <div
                    class="pf-c-select__toggle-wrapper"
                  >
                    <span
                      class="pf-c-select__toggle-text"
                    >
                      Engagement Status
                    </span>
                  </div>
                  <span
                    class="pf-c-select__toggle-arrow"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        transform=""
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <div
              class="pf-c-input-group"
            >
              <button
                aria-disabled="false"
                aria-label="search button for search input"
                class="pf-c-button pf-m-control"
                data-ouia-component-id="3"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M304 416h-64a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm-128-64h-48V48a16 16 0 0 0-16-16H80a16 16 0 0 0-16 16v304H16c-14.19 0-21.37 17.24-11.29 27.31l80 96a16 16 0 0 0 22.62 0l80-96C197.35 369.26 190.22 352 176 352zm256-192H240a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h192a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm-64 128H240a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h128a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zM496 32H240a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h256a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    transform=""
                  />
                </svg>
              </button>
              <div
                class="pf-c-select"
                data-ouia-component-id="4"
                data-ouia-component-type="PF4/Select"
                data-ouia-safe="true"
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby=" sort_dropdown"
                  class="pf-c-select__toggle"
                  id="sort_dropdown"
                  type="button"
                >
                  <div
                    class="pf-c-select__toggle-wrapper"
                  >
                    <span
                      class="pf-c-select__toggle-text"
                    >
                      Sort by
                    </span>
                  </div>
                  <span
                    class="pf-c-select__toggle-arrow"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        transform=""
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <div
            class=""
          >
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-secondary"
              data-cy="reset_button"
              data-ouia-component-id="5"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Reset
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-l-flex pf-m-justify-content-flex-start"
    >
      <div
        class="pf-l-flex pf-m-column pf-m-row-on-md"
        style="margin-top: 2rem;"
      >
        <div
          class="pf-m-grow"
        >
          <div
            class="pf-c-input-group"
          >
            <button
              aria-disabled="false"
              aria-label="search button for search input"
              class="pf-c-button pf-m-control"
              data-ouia-component-id="0"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  transform=""
                />
              </svg>
            </button>
            <input
              aria-invalid="false"
              aria-label="Search for an engagement"
              class="pf-c-form-control"
              data-cy="search_input"
              placeholder="Search for an engagement"
              style="min-width: 15rem;"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="pf-c-input-group"
          >
            <button
              aria-disabled="false"
              aria-label="search button for search input"
              class="pf-c-button pf-m-control"
              data-ouia-component-id="1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
                  transform=""
                />
              </svg>
            </button>
            <div
              class="pf-c-select"
              data-ouia-component-id="2"
              data-ouia-component-type="PF4/Select"
              data-ouia-safe="true"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=" filter_dropdown"
                class="pf-c-select__toggle"
                id="filter_dropdown"
                type="button"
              >
                <div
                  class="pf-c-select__toggle-wrapper"
                >
                  <span
                    class="pf-c-select__toggle-text"
                  >
                    Engagement Status
                  </span>
                </div>
                <span
                  class="pf-c-select__toggle-arrow"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      transform=""
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class="pf-c-input-group"
          >
            <button
              aria-disabled="false"
              aria-label="search button for search input"
              class="pf-c-button pf-m-control"
              data-ouia-component-id="3"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M304 416h-64a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm-128-64h-48V48a16 16 0 0 0-16-16H80a16 16 0 0 0-16 16v304H16c-14.19 0-21.37 17.24-11.29 27.31l80 96a16 16 0 0 0 22.62 0l80-96C197.35 369.26 190.22 352 176 352zm256-192H240a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h192a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm-64 128H240a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h128a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zM496 32H240a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h256a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                  transform=""
                />
              </svg>
            </button>
            <div
              class="pf-c-select"
              data-ouia-component-id="4"
              data-ouia-component-type="PF4/Select"
              data-ouia-safe="true"
            >
              <button
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=" sort_dropdown"
                class="pf-c-select__toggle"
                id="sort_dropdown"
                type="button"
              >
                <div
                  class="pf-c-select__toggle-wrapper"
                >
                  <span
                    class="pf-c-select__toggle-text"
                  >
                    Sort by
                  </span>
                </div>
                <span
                  class="pf-c-select__toggle-arrow"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      transform=""
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-secondary"
            data-cy="reset_button"
            data-ouia-component-id="5"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Reset
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
